WPMedia/fusion-news-theme-blocks
# TMEDIA-441: Add link to forgot password and implement block

## Description
Add Forgot Password link and block for handling reset requests.

## Jira Ticket
- [TMEDIA-441](https://arcpublishing.atlassian.net/browse/TMEDIA-441)

## Acceptance Criteria
* “Forgot Password?” link is added to existing Identity - Log In Links block
  * Under Log In section of the Identity Log In Links block, add “Show Forgot Password” checkbox with Forgot Password * Link URL with /account/forgot-password/ as the default value (just like how “Show Login link” is set up):
* Create a new block “Forgot Password” that can be dragged onto a new page ‘/account/forgot-password/’ 
* Forgot Password block uses icon from the Icon Library of 'user-question'
* Block includes Headline (Forgot Password), Subtext (Enter the email you use for your account), Email field (valid email address handled in other subtask - TMEDIA-439: ValidationIN PROGRESS) and Submit button. These should be translatable strings so we can add other languages later.
* On click of Submit button, success message shows Headline (Check your email to reset your password) and Subtext (If there is an account associated with your email address, you’ll receive a link in your inbox to reset your password.). These should be translatable strings so we can add other languages later.

## Test Steps
1. Checkout this branch `git checkout TMEDIA-441-forgot-password`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Create a new page in http://localhost/pagebuilder/pages ( Account / Forgot Password  ::  /account/forgot-password/ ) :
![image](https://user-images.githubusercontent.com/2287238/147988142-00dfedf8-848d-419a-a54b-5c48062b642a.png)
4. Use the Single Column Layout:
![image](https://user-images.githubusercontent.com/2287238/147988484-f7501b9e-2d7a-437e-a9c8-806dde2d62c4.png)
5. Add the Navigation Fragment to the navigation section:
![image](https://user-images.githubusercontent.com/2287238/147988347-37e84c3c-bf83-41fe-b4ca-edb99e362a47.png)
![image](https://user-images.githubusercontent.com/2287238/147988384-334acc70-bb51-4790-ba8d-3a88e2ce9224.png)
6. Add the Identity Forgot Password block to the main section:
![image](https://user-images.githubusercontent.com/2287238/147988554-49c2bb5b-cbf2-4923-9275-f6f54057a056.png)
7.  Publish the page
8. Enter a valid email and submit.
9. Add Login Links Block to the page and verify that checking the show link button presents the "Forgot Password?" prompt:
![image](https://user-images.githubusercontent.com/2287238/147988903-cba29676-6532-4835-82f8-3e11a507ec6a.png)
![image](https://user-images.githubusercontent.com/2287238/147988919-2a13757a-a513-4645-b828-79de89ebc882.png)

## Effect Of Changes
### Before
N/A

### After
![image](https://user-images.githubusercontent.com/2287238/147988689-5ff3e045-a47b-4274-bdae-38527c5c04c1.png)
![image](https://user-images.githubusercontent.com/2287238/147988708-8b58f29e-0dbf-45dc-81d5-30f19c0a3ce9.png)

## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
